posthog/api/person.py

```diff
@@ -41,11 +42,25 @@ class CursorPagination(BaseCursorPagination):
     page_size = 100
 
 
+class PersonFilter(filters.FilterSet):
+    email = filters.CharFilter(field_name="properties__email")
+    distinct_id = filters.CharFilter(field_name="persondistinctid__distinct_id")
+    key_identifier = filters.CharFilter(method="key_identifier_filter")
+
+    def key_identifier_filter(self, queryset, attr, *args, **kwargs):
+        """
+        Filters persons by email or distinct ID
+        """
+        return queryset.filter(Q(persondistinctid__distinct_id=args[0]) | Q(properties__email=args[0]))
```

Comment:
I'm not confident enough in my django to just approve this, so I'll ask: does this scope by team as well?

Reply:
This uses django-filter which **only adds extra filtering** to the existing queryset, as we're only further restricting the queryset, there should be no issues related to the team scoping, but I'll definitely add a test to be abs sure.